crates/stages/stages/src/stages/execution.rs

```diff
@@ -221,8 +221,9 @@ where
             provider.tx_ref(),
             provider.static_file_provider().clone(),
         ));
-        let mut executor = self.executor_provider.batch_executor(db, prune_modes);
+        let mut executor = self.executor_provider.batch_executor(db);
         executor.set_tip(max_block);
+        executor.set_prune_modes(prune_modes);
```

Comment:
the ideal solution would be not to do this at all and instead apply this to the output